cardinal/ecs/world_context.go

```diff
@@ -12,14 +12,14 @@ import (
 type WorldContext interface {
 	CurrentTick() uint64
 	Logger() *zerolog.Logger
-	NewSearch(filter Filterable) (*Search, error)
 
-	// Meant for internal ECS access only
+	// For internal use.
+	NewSearch(filter Filterable) (*Search, error)
```

Comment:
NewSearch is not just for internal use

Reply:
It has to be. Cardinal uses a completely different method. It was the entire purpose of this PR. But fine I can move it according to your preference it just a comment.